src/ORM/Association.php

```diff
@@ -518,11 +520,32 @@ public function attachTo(Query $query, array $options = [])
 
         $joinOptions = ['table' => 1, 'conditions' => 1, 'type' => 1];
         $options['conditions'] = $dummy->clause('where');
-        $query->join([$target->alias() => array_intersect_key($options, $joinOptions)]);
+        $query->join([$this->_name => array_intersect_key($options, $joinOptions)]);
```

Comment:
Ok, if they are the same then it doesn't matter. Do we want to worry about the scenario where an association is made and then the target's alias is changed?